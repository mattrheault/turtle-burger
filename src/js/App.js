import React from 'react';
import { render } from 'react-dom';
import { Provider } from 'react-redux';
import { Router } from 'react-router';
import { createBrowserHistory } from 'history';
import * as AppActions from './actions/AppActions';
import configureStore from './utils/configureStore';
import Routes from './Routes';

export default class Application {
  start() {
    this.startRedux();
    this.startRouter();
  }

  startRedux() {
    this.store = configureStore();
    this.store.dispatch(AppActions.startApp());
  }

  startRouter() {
    const history = createBrowserHistory();
    render(
      <Provider store={this.store}>
        <Router history={history}>{Routes}</Router>
      </Provider>,
      document.getElementById('root')
    );
  }
}
