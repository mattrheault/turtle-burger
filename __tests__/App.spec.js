import App from '../src/js/App';

describe('App', () => {
  it('should start redux without throwing', () => {
    let thrownErr;
    try {
      new App().startRedux();
    } catch (err) {
      thrownErr = err;
    }
    expect(thrownErr).toMatchSnapshot();
  });
});
