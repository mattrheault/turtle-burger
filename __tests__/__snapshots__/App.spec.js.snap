// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App should start redux without throwing 1`] = `undefined`;
